src/v/raft/heartbeat_manager.cc

```diff
@@ -44,49 +55,58 @@ static std::vector<heartbeat_manager::node_heartbeat> requests_for_range(
     if (c.empty()) {
         return {};
     }
+
+    // Set of follower nodes whose heartbeat_failed status indicates
+    // that we should tear down their TCP connection before next heartbeat
+    absl::flat_hash_set<model::node_id> reconnect_nodes;
+
     auto last_heartbeat = clock_type::now() - heartbeat_interval;
     for (auto& ptr : c) {
         if (!ptr->is_leader()) {
             continue;
         }
 
-        auto maybe_create_follower_request = [ptr,
-                                              last_heartbeat,
-                                              &pending_beats](
-                                               const vnode& rni) mutable {
-            // special case self beat
-            // self beat is used to make sure that the protocol will make
-            // progress when there is only on node
-            if (rni == ptr->self()) {
-                pending_beats[rni.id()].emplace_back(
-                  heartbeat_metadata{ptr->meta(), rni}, 0);
-                return;
-            }
+        auto maybe_create_follower_request =
```

Comment:
@jcsp can you split this commit into one with changes and second with formatting applied ? Now it is hard to say what have changed in this lambda.